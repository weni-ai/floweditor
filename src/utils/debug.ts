import { FlowStoreProps } from 'components/flow/Flow';
import { DebugState } from 'store/editor';

import mutate from 'immutability-helper';

/* istanbul ignore next -- @preserve */
export default class Debug {
  private props: FlowStoreProps;
  private state: DebugState;

  constructor(props: FlowStoreProps, initial: DebugState) {
    this.props = props;
    this.state = initial || { showUUIDs: false };
  }

  public showUUIDs(): DebugState {
    const updated = mutate(this.state, { $merge: { showUUIDs: true } });
    this.props.mergeEditorState({ debug: updated });
    return updated;
  }
}
