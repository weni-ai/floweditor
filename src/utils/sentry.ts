import * as Sentry from '@sentry/react';

const initSentry = (dsn: string) => {
  Sentry.init({
    dsn,
    integrations: [
      Sentry.browserTracingIntegration(),
      Sentry.httpClientIntegration({
        failedRequestStatusCodes: [[400, 499], [500, 599]],
      }),
      Sentry.replayIntegration(),
      Sentry.replayCanvasIntegration(),
    ],
    tracesSampleRate: 1.0,
    replaysSessionSampleRate: 0.1,
    replaysOnErrorSampleRate: 1.0,
  });
};

export default initSentry;
