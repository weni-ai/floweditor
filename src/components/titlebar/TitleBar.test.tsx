import TitleBar, {
  moveIconSpecId,
  removeIconSpecId,
  TitleBarProps,
} from 'components/titlebar/TitleBar';
import React from 'react';
import { act, fireEvent, render } from 'test/utils';

const baseProps: TitleBarProps = {
  title: 'Send Message',
  onRemoval: vi.fn(),
};

describe(TitleBar.name, () => {
  describe('render', () => {
    it('should render self, children with base props', () => {
      const { container } = render(<TitleBar {...baseProps} />);
      expect(container).toMatchSnapshot();
    });

    it('should apply _className prop', () => {
      const { container } = render(
        <TitleBar {...baseProps} __className="some-classy-class" />,
      );
      expect(container).toMatchSnapshot();
    });

    describe('move icon', () => {
      it('should render move icon', () => {
        const { container } = render(
          <TitleBar {...baseProps} showMove={true} />,
        );
        expect(container).toMatchSnapshot();
      });

      it('should call onMoveUp', () => {
        const onMoveUp = vi.fn();
        const { baseElement, getByTestId } = render(
          <TitleBar {...baseProps} showMove={true} onMoveUp={onMoveUp} />,
        );
        expect(baseElement).toMatchSnapshot();

        // click on the move up button
        fireEvent.mouseUp(getByTestId(moveIconSpecId));
        expect(onMoveUp).toHaveBeenCalledTimes(1);
      });
    });

    describe('remove icon', () => {
      it('should render remove icon', () => {
        const { baseElement } = render(
          <TitleBar {...baseProps} showMove={true} showRemoval={true} />,
        );
        expect(baseElement).toMatchSnapshot();
      });
    });

    describe('confirmation', () => {
      it('should render confirmation markup and call onRemoval prop', async () => {
        const handleRemoveClicked = vi.fn();
        const { baseElement, getByTestId, getByText } = render(
          <TitleBar
            {...baseProps}
            showMove={true}
            showRemoval={true}
            onRemoval={handleRemoveClicked}
          />,
        );

        expect(baseElement).toMatchSnapshot();

        await act(async () => {
          fireEvent.mouseUp(getByTestId(removeIconSpecId));
        });
        fireEvent.click(getByText('Confirm'));
        expect(handleRemoveClicked).toHaveBeenCalledTimes(1);
      });
    });
  });
});
