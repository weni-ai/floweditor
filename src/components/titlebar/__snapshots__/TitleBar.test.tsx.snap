// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TitleBar > render > confirmation > should render confirmation markup and call onRemoval prop 1`] = `
<body>
  <div>
    <div
      class="_titlebar_122edd null"
      data-spec="titlebar-container"
    >
      <div
        class="_normal_122edd undefined null"
        data-spec="titlebar"
      >
        <div
          class="_up_button_122edd"
          data-testid="move-icon"
        >
          <div
            data-use-vue-component-wrap=""
            data-v-app=""
            style="all: unset;"
          >
            <svg
              class="unnnic-icon unnnic-icon__size--xs unnnic-icon-scheme--neutral-snow _up_button_122edd"
              data-spec="move"
              data-v-edf1acd8=""
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="primary"
                d="M34.6851 25.9913L20.9567 10.4043C20.8323 10.2746 20.683 10.1714 20.5177 10.1009C20.3524 10.0304 20.1746 9.99402 19.9949 9.99402C19.8152 9.99402 19.6374 10.0304 19.4721 10.1009C19.3068 10.1714 19.1575 10.2746 19.0332 10.4043L5.311 25.9913C5.20125 26.1149 5.11748 26.2594 5.06465 26.416C5.01183 26.5727 4.99104 26.7384 5.00352 26.9032C5.01599 27.0681 5.06148 27.2288 5.13727 27.3757C5.21307 27.5226 5.31763 27.6528 5.44474 27.7586L7.75697 29.7046C8.01481 29.922 8.34778 30.0292 8.68402 30.0033C9.02027 29.9773 9.33283 29.8202 9.55426 29.5658L19.7593 17.751C19.7886 17.7169 19.825 17.6895 19.8659 17.6708C19.9067 17.652 19.9512 17.6423 19.9962 17.6423C20.0411 17.6423 20.0856 17.652 20.1265 17.6708C20.1673 17.6895 20.2037 17.7169 20.233 17.751L30.4418 29.5621C30.5518 29.6887 30.6857 29.7924 30.8358 29.8673C30.986 29.9421 31.1494 29.9867 31.3167 29.9983C31.4831 30.0131 31.6508 29.9944 31.8099 29.9434C31.969 29.8923 32.1162 29.8098 32.2429 29.7008L34.5563 27.7536C34.6832 27.6476 34.7874 27.5174 34.863 27.3704C34.9386 27.2235 34.984 27.0629 34.9965 26.8982C35.0089 26.7334 34.9883 26.5678 34.9356 26.4112C34.883 26.2546 34.7995 26.1101 34.6901 25.9863L34.6851 25.9913Z"
                fill="#3B414D"
              />
            </svg>
          </div>
        </div>
        <div
          class="_titletext_122edd u font secondary body-md bold"
        >
          Send Message
        </div>
        <div
          class="_remove_button_122edd"
          data-testid="remove-icon"
        >
          <div
            data-use-vue-component-wrap=""
            data-v-app=""
            style="all: unset;"
          >
            <svg
              class="unnnic-icon unnnic-icon__size--xs unnnic-icon-scheme--neutral-snow"
              data-v-edf1acd8=""
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="primary"
                d="M34.0592 34.9982C33.9361 34.9985 33.8142 34.9744 33.7004 34.9272C33.5867 34.88 33.4835 34.8107 33.3969 34.7232L19.9995 21.3271L6.60213 34.7232C6.51561 34.811 6.4125 34.8808 6.29879 34.9284C6.18507 34.976 6.06303 35.0005 5.93976 35.0005C5.81648 35.0005 5.69444 34.976 5.58073 34.9284C5.46701 34.8808 5.3639 34.811 5.27739 34.7232C5.10209 34.5471 5.00367 34.3087 5.00367 34.0602C5.00367 33.8117 5.10209 33.5734 5.27739 33.3972L18.6748 19.9999L5.27739 6.60374C5.09992 6.42753 5.00244 6.19132 5.00244 5.94137C5.00244 5.69142 5.09992 5.45522 5.27739 5.279C5.4536 5.10154 5.68981 5.00406 5.93976 5.00406C6.18971 5.00406 6.42591 5.10154 6.60213 5.279L19.9995 18.6751L33.3969 5.27775C33.4834 5.18994 33.5865 5.1202 33.7002 5.0726C33.8139 5.025 33.936 5.00049 34.0592 5.00049C34.1825 5.00049 34.3046 5.025 34.4183 5.0726C34.532 5.1202 34.6351 5.18994 34.7216 5.27775C34.8092 5.36454 34.8787 5.46782 34.9261 5.58161C34.9735 5.69541 34.9979 5.81747 34.9979 5.94075C34.9979 6.06403 34.9735 6.18609 34.9261 6.29988C34.8787 6.41367 34.8092 6.51695 34.7216 6.60374L21.3242 20.0011L34.7216 33.3985C34.8969 33.5746 34.9953 33.813 34.9953 34.0615C34.9953 34.31 34.8969 34.5484 34.7216 34.7245C34.6347 34.8115 34.5315 34.8805 34.4178 34.9275C34.3041 34.9745 34.1822 34.9985 34.0592 34.9982Z"
                fill="#3B414D"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`TitleBar > render > move icon > should call onMoveUp 1`] = `
<body>
  <div>
    <div
      class="_titlebar_122edd null"
      data-spec="titlebar-container"
    >
      <div
        class="_normal_122edd undefined null"
        data-spec="titlebar"
      >
        <div
          class="_up_button_122edd"
          data-testid="move-icon"
        >
          <div
            data-use-vue-component-wrap=""
            data-v-app=""
            style="all: unset;"
          >
            <svg
              class="unnnic-icon unnnic-icon__size--xs unnnic-icon-scheme--neutral-snow _up_button_122edd"
              data-spec="move"
              data-v-edf1acd8=""
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="primary"
                d="M34.6851 25.9913L20.9567 10.4043C20.8323 10.2746 20.683 10.1714 20.5177 10.1009C20.3524 10.0304 20.1746 9.99402 19.9949 9.99402C19.8152 9.99402 19.6374 10.0304 19.4721 10.1009C19.3068 10.1714 19.1575 10.2746 19.0332 10.4043L5.311 25.9913C5.20125 26.1149 5.11748 26.2594 5.06465 26.416C5.01183 26.5727 4.99104 26.7384 5.00352 26.9032C5.01599 27.0681 5.06148 27.2288 5.13727 27.3757C5.21307 27.5226 5.31763 27.6528 5.44474 27.7586L7.75697 29.7046C8.01481 29.922 8.34778 30.0292 8.68402 30.0033C9.02027 29.9773 9.33283 29.8202 9.55426 29.5658L19.7593 17.751C19.7886 17.7169 19.825 17.6895 19.8659 17.6708C19.9067 17.652 19.9512 17.6423 19.9962 17.6423C20.0411 17.6423 20.0856 17.652 20.1265 17.6708C20.1673 17.6895 20.2037 17.7169 20.233 17.751L30.4418 29.5621C30.5518 29.6887 30.6857 29.7924 30.8358 29.8673C30.986 29.9421 31.1494 29.9867 31.3167 29.9983C31.4831 30.0131 31.6508 29.9944 31.8099 29.9434C31.969 29.8923 32.1162 29.8098 32.2429 29.7008L34.5563 27.7536C34.6832 27.6476 34.7874 27.5174 34.863 27.3704C34.9386 27.2235 34.984 27.0629 34.9965 26.8982C35.0089 26.7334 34.9883 26.5678 34.9356 26.4112C34.883 26.2546 34.7995 26.1101 34.6901 25.9863L34.6851 25.9913Z"
                fill="#3B414D"
              />
            </svg>
          </div>
        </div>
        <div
          class="_titletext_122edd u font secondary body-md bold"
        >
          Send Message
        </div>
        <div
          class="_remove_button_122edd"
          data-testid="remove-icon"
        >
          <div
            data-use-vue-component-wrap=""
            data-v-app=""
            style="all: unset;"
          >
            <svg
              class="unnnic-icon unnnic-icon__size--xs unnnic-icon-scheme--neutral-snow"
              data-v-edf1acd8=""
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="primary"
                d="M34.0592 34.9982C33.9361 34.9985 33.8142 34.9744 33.7004 34.9272C33.5867 34.88 33.4835 34.8107 33.3969 34.7232L19.9995 21.3271L6.60213 34.7232C6.51561 34.811 6.4125 34.8808 6.29879 34.9284C6.18507 34.976 6.06303 35.0005 5.93976 35.0005C5.81648 35.0005 5.69444 34.976 5.58073 34.9284C5.46701 34.8808 5.3639 34.811 5.27739 34.7232C5.10209 34.5471 5.00367 34.3087 5.00367 34.0602C5.00367 33.8117 5.10209 33.5734 5.27739 33.3972L18.6748 19.9999L5.27739 6.60374C5.09992 6.42753 5.00244 6.19132 5.00244 5.94137C5.00244 5.69142 5.09992 5.45522 5.27739 5.279C5.4536 5.10154 5.68981 5.00406 5.93976 5.00406C6.18971 5.00406 6.42591 5.10154 6.60213 5.279L19.9995 18.6751L33.3969 5.27775C33.4834 5.18994 33.5865 5.1202 33.7002 5.0726C33.8139 5.025 33.936 5.00049 34.0592 5.00049C34.1825 5.00049 34.3046 5.025 34.4183 5.0726C34.532 5.1202 34.6351 5.18994 34.7216 5.27775C34.8092 5.36454 34.8787 5.46782 34.9261 5.58161C34.9735 5.69541 34.9979 5.81747 34.9979 5.94075C34.9979 6.06403 34.9735 6.18609 34.9261 6.29988C34.8787 6.41367 34.8092 6.51695 34.7216 6.60374L21.3242 20.0011L34.7216 33.3985C34.8969 33.5746 34.9953 33.813 34.9953 34.0615C34.9953 34.31 34.8969 34.5484 34.7216 34.7245C34.6347 34.8115 34.5315 34.8805 34.4178 34.9275C34.3041 34.9745 34.1822 34.9985 34.0592 34.9982Z"
                fill="#3B414D"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`TitleBar > render > move icon > should render move icon 1`] = `
<div>
  <div
    class="_titlebar_122edd null"
    data-spec="titlebar-container"
  >
    <div
      class="_normal_122edd undefined null"
      data-spec="titlebar"
    >
      <div
        class="_up_button_122edd"
        data-testid="move-icon"
      >
        <div
          data-use-vue-component-wrap=""
          data-v-app=""
          style="all: unset;"
        >
          <svg
            class="unnnic-icon unnnic-icon__size--xs unnnic-icon-scheme--neutral-snow _up_button_122edd"
            data-spec="move"
            data-v-edf1acd8=""
            fill="none"
            height="40"
            viewBox="0 0 40 40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="primary"
              d="M34.6851 25.9913L20.9567 10.4043C20.8323 10.2746 20.683 10.1714 20.5177 10.1009C20.3524 10.0304 20.1746 9.99402 19.9949 9.99402C19.8152 9.99402 19.6374 10.0304 19.4721 10.1009C19.3068 10.1714 19.1575 10.2746 19.0332 10.4043L5.311 25.9913C5.20125 26.1149 5.11748 26.2594 5.06465 26.416C5.01183 26.5727 4.99104 26.7384 5.00352 26.9032C5.01599 27.0681 5.06148 27.2288 5.13727 27.3757C5.21307 27.5226 5.31763 27.6528 5.44474 27.7586L7.75697 29.7046C8.01481 29.922 8.34778 30.0292 8.68402 30.0033C9.02027 29.9773 9.33283 29.8202 9.55426 29.5658L19.7593 17.751C19.7886 17.7169 19.825 17.6895 19.8659 17.6708C19.9067 17.652 19.9512 17.6423 19.9962 17.6423C20.0411 17.6423 20.0856 17.652 20.1265 17.6708C20.1673 17.6895 20.2037 17.7169 20.233 17.751L30.4418 29.5621C30.5518 29.6887 30.6857 29.7924 30.8358 29.8673C30.986 29.9421 31.1494 29.9867 31.3167 29.9983C31.4831 30.0131 31.6508 29.9944 31.8099 29.9434C31.969 29.8923 32.1162 29.8098 32.2429 29.7008L34.5563 27.7536C34.6832 27.6476 34.7874 27.5174 34.863 27.3704C34.9386 27.2235 34.984 27.0629 34.9965 26.8982C35.0089 26.7334 34.9883 26.5678 34.9356 26.4112C34.883 26.2546 34.7995 26.1101 34.6901 25.9863L34.6851 25.9913Z"
              fill="#3B414D"
            />
          </svg>
        </div>
      </div>
      <div
        class="_titletext_122edd u font secondary body-md bold"
      >
        Send Message
      </div>
      <div
        class="_remove_button_122edd"
        data-testid="remove-icon"
      >
        <div
          data-use-vue-component-wrap=""
          data-v-app=""
          style="all: unset;"
        >
          <svg
            class="unnnic-icon unnnic-icon__size--xs unnnic-icon-scheme--neutral-snow"
            data-v-edf1acd8=""
            fill="none"
            height="40"
            viewBox="0 0 40 40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="primary"
              d="M34.0592 34.9982C33.9361 34.9985 33.8142 34.9744 33.7004 34.9272C33.5867 34.88 33.4835 34.8107 33.3969 34.7232L19.9995 21.3271L6.60213 34.7232C6.51561 34.811 6.4125 34.8808 6.29879 34.9284C6.18507 34.976 6.06303 35.0005 5.93976 35.0005C5.81648 35.0005 5.69444 34.976 5.58073 34.9284C5.46701 34.8808 5.3639 34.811 5.27739 34.7232C5.10209 34.5471 5.00367 34.3087 5.00367 34.0602C5.00367 33.8117 5.10209 33.5734 5.27739 33.3972L18.6748 19.9999L5.27739 6.60374C5.09992 6.42753 5.00244 6.19132 5.00244 5.94137C5.00244 5.69142 5.09992 5.45522 5.27739 5.279C5.4536 5.10154 5.68981 5.00406 5.93976 5.00406C6.18971 5.00406 6.42591 5.10154 6.60213 5.279L19.9995 18.6751L33.3969 5.27775C33.4834 5.18994 33.5865 5.1202 33.7002 5.0726C33.8139 5.025 33.936 5.00049 34.0592 5.00049C34.1825 5.00049 34.3046 5.025 34.4183 5.0726C34.532 5.1202 34.6351 5.18994 34.7216 5.27775C34.8092 5.36454 34.8787 5.46782 34.9261 5.58161C34.9735 5.69541 34.9979 5.81747 34.9979 5.94075C34.9979 6.06403 34.9735 6.18609 34.9261 6.29988C34.8787 6.41367 34.8092 6.51695 34.7216 6.60374L21.3242 20.0011L34.7216 33.3985C34.8969 33.5746 34.9953 33.813 34.9953 34.0615C34.9953 34.31 34.8969 34.5484 34.7216 34.7245C34.6347 34.8115 34.5315 34.8805 34.4178 34.9275C34.3041 34.9745 34.1822 34.9985 34.0592 34.9982Z"
              fill="#3B414D"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TitleBar > render > remove icon > should render remove icon 1`] = `
<body>
  <div>
    <div
      class="_titlebar_122edd null"
      data-spec="titlebar-container"
    >
      <div
        class="_normal_122edd undefined null"
        data-spec="titlebar"
      >
        <div
          class="_up_button_122edd"
          data-testid="move-icon"
        >
          <div
            data-use-vue-component-wrap=""
            data-v-app=""
            style="all: unset;"
          >
            <svg
              class="unnnic-icon unnnic-icon__size--xs unnnic-icon-scheme--neutral-snow _up_button_122edd"
              data-spec="move"
              data-v-edf1acd8=""
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="primary"
                d="M34.6851 25.9913L20.9567 10.4043C20.8323 10.2746 20.683 10.1714 20.5177 10.1009C20.3524 10.0304 20.1746 9.99402 19.9949 9.99402C19.8152 9.99402 19.6374 10.0304 19.4721 10.1009C19.3068 10.1714 19.1575 10.2746 19.0332 10.4043L5.311 25.9913C5.20125 26.1149 5.11748 26.2594 5.06465 26.416C5.01183 26.5727 4.99104 26.7384 5.00352 26.9032C5.01599 27.0681 5.06148 27.2288 5.13727 27.3757C5.21307 27.5226 5.31763 27.6528 5.44474 27.7586L7.75697 29.7046C8.01481 29.922 8.34778 30.0292 8.68402 30.0033C9.02027 29.9773 9.33283 29.8202 9.55426 29.5658L19.7593 17.751C19.7886 17.7169 19.825 17.6895 19.8659 17.6708C19.9067 17.652 19.9512 17.6423 19.9962 17.6423C20.0411 17.6423 20.0856 17.652 20.1265 17.6708C20.1673 17.6895 20.2037 17.7169 20.233 17.751L30.4418 29.5621C30.5518 29.6887 30.6857 29.7924 30.8358 29.8673C30.986 29.9421 31.1494 29.9867 31.3167 29.9983C31.4831 30.0131 31.6508 29.9944 31.8099 29.9434C31.969 29.8923 32.1162 29.8098 32.2429 29.7008L34.5563 27.7536C34.6832 27.6476 34.7874 27.5174 34.863 27.3704C34.9386 27.2235 34.984 27.0629 34.9965 26.8982C35.0089 26.7334 34.9883 26.5678 34.9356 26.4112C34.883 26.2546 34.7995 26.1101 34.6901 25.9863L34.6851 25.9913Z"
                fill="#3B414D"
              />
            </svg>
          </div>
        </div>
        <div
          class="_titletext_122edd u font secondary body-md bold"
        >
          Send Message
        </div>
        <div
          class="_remove_button_122edd"
          data-testid="remove-icon"
        >
          <div
            data-use-vue-component-wrap=""
            data-v-app=""
            style="all: unset;"
          >
            <svg
              class="unnnic-icon unnnic-icon__size--xs unnnic-icon-scheme--neutral-snow"
              data-v-edf1acd8=""
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="primary"
                d="M34.0592 34.9982C33.9361 34.9985 33.8142 34.9744 33.7004 34.9272C33.5867 34.88 33.4835 34.8107 33.3969 34.7232L19.9995 21.3271L6.60213 34.7232C6.51561 34.811 6.4125 34.8808 6.29879 34.9284C6.18507 34.976 6.06303 35.0005 5.93976 35.0005C5.81648 35.0005 5.69444 34.976 5.58073 34.9284C5.46701 34.8808 5.3639 34.811 5.27739 34.7232C5.10209 34.5471 5.00367 34.3087 5.00367 34.0602C5.00367 33.8117 5.10209 33.5734 5.27739 33.3972L18.6748 19.9999L5.27739 6.60374C5.09992 6.42753 5.00244 6.19132 5.00244 5.94137C5.00244 5.69142 5.09992 5.45522 5.27739 5.279C5.4536 5.10154 5.68981 5.00406 5.93976 5.00406C6.18971 5.00406 6.42591 5.10154 6.60213 5.279L19.9995 18.6751L33.3969 5.27775C33.4834 5.18994 33.5865 5.1202 33.7002 5.0726C33.8139 5.025 33.936 5.00049 34.0592 5.00049C34.1825 5.00049 34.3046 5.025 34.4183 5.0726C34.532 5.1202 34.6351 5.18994 34.7216 5.27775C34.8092 5.36454 34.8787 5.46782 34.9261 5.58161C34.9735 5.69541 34.9979 5.81747 34.9979 5.94075C34.9979 6.06403 34.9735 6.18609 34.9261 6.29988C34.8787 6.41367 34.8092 6.51695 34.7216 6.60374L21.3242 20.0011L34.7216 33.3985C34.8969 33.5746 34.9953 33.813 34.9953 34.0615C34.9953 34.31 34.8969 34.5484 34.7216 34.7245C34.6347 34.8115 34.5315 34.8805 34.4178 34.9275C34.3041 34.9745 34.1822 34.9985 34.0592 34.9982Z"
                fill="#3B414D"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`TitleBar > render > should apply _className prop 1`] = `
<div>
  <div
    class="_titlebar_122edd null"
    data-spec="titlebar-container"
  >
    <div
      class="_normal_122edd some-classy-class null"
      data-spec="titlebar"
    >
      <div
        class="_up_button_122edd"
        data-spec="move"
      />
      <div
        class="_titletext_122edd u font secondary body-md bold"
      >
        Send Message
      </div>
      <div
        class="_remove_button_122edd"
        data-testid="remove-icon"
      >
        <div
          data-use-vue-component-wrap=""
          data-v-app=""
          style="all: unset;"
        >
          <svg
            class="unnnic-icon unnnic-icon__size--xs unnnic-icon-scheme--neutral-snow"
            data-v-edf1acd8=""
            fill="none"
            height="40"
            viewBox="0 0 40 40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="primary"
              d="M34.0592 34.9982C33.9361 34.9985 33.8142 34.9744 33.7004 34.9272C33.5867 34.88 33.4835 34.8107 33.3969 34.7232L19.9995 21.3271L6.60213 34.7232C6.51561 34.811 6.4125 34.8808 6.29879 34.9284C6.18507 34.976 6.06303 35.0005 5.93976 35.0005C5.81648 35.0005 5.69444 34.976 5.58073 34.9284C5.46701 34.8808 5.3639 34.811 5.27739 34.7232C5.10209 34.5471 5.00367 34.3087 5.00367 34.0602C5.00367 33.8117 5.10209 33.5734 5.27739 33.3972L18.6748 19.9999L5.27739 6.60374C5.09992 6.42753 5.00244 6.19132 5.00244 5.94137C5.00244 5.69142 5.09992 5.45522 5.27739 5.279C5.4536 5.10154 5.68981 5.00406 5.93976 5.00406C6.18971 5.00406 6.42591 5.10154 6.60213 5.279L19.9995 18.6751L33.3969 5.27775C33.4834 5.18994 33.5865 5.1202 33.7002 5.0726C33.8139 5.025 33.936 5.00049 34.0592 5.00049C34.1825 5.00049 34.3046 5.025 34.4183 5.0726C34.532 5.1202 34.6351 5.18994 34.7216 5.27775C34.8092 5.36454 34.8787 5.46782 34.9261 5.58161C34.9735 5.69541 34.9979 5.81747 34.9979 5.94075C34.9979 6.06403 34.9735 6.18609 34.9261 6.29988C34.8787 6.41367 34.8092 6.51695 34.7216 6.60374L21.3242 20.0011L34.7216 33.3985C34.8969 33.5746 34.9953 33.813 34.9953 34.0615C34.9953 34.31 34.8969 34.5484 34.7216 34.7245C34.6347 34.8115 34.5315 34.8805 34.4178 34.9275C34.3041 34.9745 34.1822 34.9985 34.0592 34.9982Z"
              fill="#3B414D"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TitleBar > render > should render self, children with base props 1`] = `
<div>
  <div
    class="_titlebar_122edd null"
    data-spec="titlebar-container"
  >
    <div
      class="_normal_122edd undefined null"
      data-spec="titlebar"
    >
      <div
        class="_up_button_122edd"
        data-spec="move"
      />
      <div
        class="_titletext_122edd u font secondary body-md bold"
      >
        Send Message
      </div>
      <div
        class="_remove_button_122edd"
        data-testid="remove-icon"
      >
        <div
          data-use-vue-component-wrap=""
          data-v-app=""
          style="all: unset;"
        >
          <svg
            class="unnnic-icon unnnic-icon__size--xs unnnic-icon-scheme--neutral-snow"
            data-v-edf1acd8=""
            fill="none"
            height="40"
            viewBox="0 0 40 40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="primary"
              d="M34.0592 34.9982C33.9361 34.9985 33.8142 34.9744 33.7004 34.9272C33.5867 34.88 33.4835 34.8107 33.3969 34.7232L19.9995 21.3271L6.60213 34.7232C6.51561 34.811 6.4125 34.8808 6.29879 34.9284C6.18507 34.976 6.06303 35.0005 5.93976 35.0005C5.81648 35.0005 5.69444 34.976 5.58073 34.9284C5.46701 34.8808 5.3639 34.811 5.27739 34.7232C5.10209 34.5471 5.00367 34.3087 5.00367 34.0602C5.00367 33.8117 5.10209 33.5734 5.27739 33.3972L18.6748 19.9999L5.27739 6.60374C5.09992 6.42753 5.00244 6.19132 5.00244 5.94137C5.00244 5.69142 5.09992 5.45522 5.27739 5.279C5.4536 5.10154 5.68981 5.00406 5.93976 5.00406C6.18971 5.00406 6.42591 5.10154 6.60213 5.279L19.9995 18.6751L33.3969 5.27775C33.4834 5.18994 33.5865 5.1202 33.7002 5.0726C33.8139 5.025 33.936 5.00049 34.0592 5.00049C34.1825 5.00049 34.3046 5.025 34.4183 5.0726C34.532 5.1202 34.6351 5.18994 34.7216 5.27775C34.8092 5.36454 34.8787 5.46782 34.9261 5.58161C34.9735 5.69541 34.9979 5.81747 34.9979 5.94075C34.9979 6.06403 34.9735 6.18609 34.9261 6.29988C34.8787 6.41367 34.8092 6.51695 34.7216 6.60374L21.3242 20.0011L34.7216 33.3985C34.8969 33.5746 34.9953 33.813 34.9953 34.0615C34.9953 34.31 34.8969 34.5484 34.7216 34.7245C34.6347 34.8115 34.5315 34.8805 34.4178 34.9275C34.3041 34.9745 34.1822 34.9985 34.0592 34.9982Z"
              fill="#3B414D"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
